godot-macros/src/class/data_models/func.rs

```diff
@@ -292,7 +325,14 @@ pub(crate) fn get_signature_info(signature: &venial::Function, has_gd_self: bool
                 } else {
                     arg.name.clone()
                 };
-                let ty = arg.ty.clone();
+                let ty = venial::TyExpr {
+                    tokens: arg
+                        .ty
+                        .tokens
+                        .iter()
+                        .map(|tt| map_self_to_class_name(tt, class_name))
+                        .collect(),
+                };
```

Comment:
I might be able to pull out all 3 usages of the basic pattern into a function that takes an `impl IntoIterator` and returns a `FromIterator`? I'll mess around with that later today, otherwise the `TyExpr` specific one should be pretty easy.

Reply:
Using iterators sounds good, too. There's also the [`quote::ToTokens`](https://docs.rs/quote/latest/quote/trait.ToTokens.html#method.to_token_stream) trait with `into_token_stream()`, but it seems not implemented for `Vec<TokenTree>`, so not sure if it would be directly applicable.